### PR TITLE
functions: Add CONCAT(...) function.

### PR DESCRIPTION
**Pull request by ajnavarro**
Signed-off-by: Antonio Jesus Navarro Perez <antnavper@gmail.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,12 +51,12 @@ We are continuously adding more functionality to go-mysql-server. We support a s
...
-- `IS_BINARY(blob)`: returns whether a BLOB is a binary file or not.
-- `SUBSTRING(str,pos)`, ` SUBSTRING(str,pos,len)`: return a substring from the provided string.
+- `IS_BINARY(blob)`: Returns whether a BLOB is a binary file or not.
+- `SUBSTRING(str,pos)`, `SUBSTRING(str,pos,len)`: Return a substring from the provided string.
 - Date and Timestamp functions: `YEAR(date)`, `MONTH(date)`, `DAY(date)`, `HOUR(date)`, `MINUTE(date)`, `SECOND(date)`, `DAYOFYEAR(date)`.
 - `ARRAY_LENGTH(json)`: If the json representation is an array, this function returns its size.
-- `SPLIT(str,sep)`: receives a string and a delimiter and returns the parts of the string splitted by the delimiter as a JSON array of strings.
-
+- `SPLIT(str,sep)`: Receives a string and a delimiter and returns the parts of the string splitted by the delimiter as a JSON array of strings.
+- `CONCAT(...)`: Concatenate any group of fields into a string one.
```
Comment (erizocosmico) [inline README.md]: s/string one/single string/

File: README.md [hunk 2]
```diff
@@ -51,12 +51,12 @@ We are continuously adding more functionality to go-mysql-server. We support a s
...
 
-- `IS_BINARY(blob)`: returns whether a BLOB is a binary file or not.
-- `SUBSTRING(str,pos)`, ` SUBSTRING(str,pos,len)`: return a substring from the provided string.
+- `IS_BINARY(blob)`: Returns whether a BLOB is a binary file or not.
+- `SUBSTRING(str,pos)`, `SUBSTRING(str,pos,len)`: Return a substring from the provided string.
 - Date and Timestamp functions: `YEAR(date)`, `MONTH(date)`, `DAY(date)`, `HOUR(date)`, `MINUTE(date)`, `SECOND(date)`, `DAYOFYEAR(date)`.
 - `ARRAY_LENGTH(json)`: If the json representation is an array, this function returns its size.
-- `SPLIT(str,sep)`: receives a string and a delimiter and returns the parts of the string splitted by the delimiter as a JSON array of strings.
-
+- `SPLIT(str,sep)`: Receives a string and a delimiter and returns the parts of the string splitted by the delimiter as a JSON array of strings.
```
Comment (kuba--) [inline README.md]: afaik `split` is irregular so past form is also `split`

if we named parameter `sep` so maybe we should say: 
> "...split by the separator"

Comment (ajnavarro) [reply README.md]: shall we change this on another PR?

Comment (kuba--) [reply README.md]: no, I just meant to say sth. like:
> Receives a string and a separator and returns the parts of the string split by the separator as a JSON array of strings.

**Comment by ajnavarro** [comment]
@kuba-- done

**Pull request merged**